coral-spark/src/main/java/com/linkedin/coral/spark/CoralToSparkSqlCallTransformationUtils.java

```diff
@@ -0,0 +1,124 @@
+
+  private static SqlCall getTransformedUnnestSqlCall(SqlCall sqlCall) {
+    final SqlNode potentialIfCall = getOrDefault(sqlCall.getOperandList(), 0, null);
+
+    if (potentialIfCall instanceof SqlCall
+        && ((SqlCall) potentialIfCall).getOperator().getName().equalsIgnoreCase("if")) {
+      final SqlNode unnestColumn = ((SqlCall) potentialIfCall).getOperandList().get(1);
+      return sqlCall.getOperator().createCall(SqlParserPos.ZERO, unnestColumn);
+    }
+    return sqlCall;
+  }
+
+  private static SqlCall getTransformedAsSqlCall(SqlCall sqlCall) {
+    if (sqlCall.operandCount() <= 2 || !(sqlCall.operand(0) instanceof SqlBasicCall)
+        || (sqlCall.operand(0) instanceof SqlBasicCall && sqlCall.operand(0).getKind() == SqlKind.VALUES)) {
+      return sqlCall;
+    }
+
+    SqlCall explodeCall = sqlCall.operand(0);
+    final SqlOperator operator = explodeCall.getOperator();
+    List<SqlNode> aliasOperands = sqlCall.getOperandList();
+
+    List<SqlNode> asOperands = new ArrayList<>();
+    asOperands.add(explodeCall);
+
+    // For POSEXPLODE case, we reorder the operands again from `val, pos` to `pos, val` to undo the reordering during creation of CoralRelNode
+    if (operator instanceof HivePosExplodeOperator && sqlCall.getOperandList().size() == 4) {
+      asOperands.add(aliasOperands.get(1));
+      asOperands.add(aliasOperands.get(3));
+      asOperands.add(aliasOperands.get(2));
+    } else {
+      asOperands.addAll(aliasOperands.subList(1, aliasOperands.size()));
+    }
+
+    return SqlLateralViewAsOperator.instance.createCall(SqlParserPos.ZERO, asOperands);
+  }
+
+  private static SqlCall getEmptyOperatorSqlCall(SqlCall sqlCall) {
+    return SqlEmptyOperator.EMPTY_OPERATOR.createCall(SqlParserPos.ZERO, sqlCall.getOperandList());
+  }
+
+  private static SqlCall getTransformedJoinSqlCall(SqlCall sqlCall) {
+    if (((SqlJoin) sqlCall).getJoinType() != JoinType.COMMA
+        || ((SqlCall) ((SqlJoin) sqlCall).getRight()).getOperator().kind != SqlKind.LATERAL) {
+      return sqlCall;
+    }
+
+    boolean isOuter = isCorrelateRightChildOuter(sqlCall);
+
+    return new SqlLateralJoin(SqlParserPos.ZERO, ((SqlJoin) sqlCall).getLeft(),
+        SqlLiteral.createBoolean(false, SqlParserPos.ZERO), JoinType.COMMA.symbol(SqlParserPos.ZERO),
+        ((SqlJoin) sqlCall).getRight(), JoinConditionType.NONE.symbol(SqlParserPos.ZERO), null, isOuter);
+  }
+
+  private static boolean isCorrelateRightChildOuter(SqlCall sqlCall) {
+    SqlCall lateralNode = (SqlCall) ((SqlJoin) sqlCall).getRight();
+    SqlCall asNode = lateralNode.operand(0);
+    SqlCall unnestNode = asNode.operand(0);
+
+    if (unnestNode.getKind() == SqlKind.UNNEST) {
+      SqlBasicCall unnestCall = (SqlBasicCall) unnestNode;
+      SqlNode ifNode = getOrDefault(unnestCall.getOperandList(), 0, null);
+      if (ifNode instanceof SqlBasicCall && ((SqlBasicCall) ifNode).getOperator().getName().equalsIgnoreCase("if")) {
+        return true;
+      }
+    }
```

Comment:
It only verifies if it has `if` child, but didn't verify if it has `ARRAY(NULL)` as the else result?

Reply:
do we need all the sub-clauses to be verified? 
is there a sample SQL use-case where we might encounter "if" child without "ARRAY(NULL)" inside the unnest operator? If not, I think the current implementation should be fine.

Reply:
I can't find such a production use case, but I think it could happen in theory?

Reply:
incorporated this change